src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -737,8 +736,8 @@ private void deleteFiles(long minimumBytesToBeDeleted, Path directoryPath, Patte
     @Override
     @SuppressWarnings("PMD.UselessOverridingMethod")
     public void shutdown() throws InterruptedException {
-        // Need to override the function for tests.
         super.shutdown();
+        isCurrentlyUploading.set(false);
```

Comment:
What will happen if the logs have already been uploaded and the Log Manager is shutting down that time? The logs would have reached CW but we would not have updated the state in LogManager. So when LogManager starts up again, it will again reupload those logs which have already been uploaded. 
This is just an optimization, but should we wait until the current execution of logs has been completed before shutting the component down? We did something similar in Shadow Manager as well.

Reply:
We could try that, but doesn't seem straightforward and will definitely need more testing. So, it'll be taken up separately as a follow up to push this out soon